src/main/java/uk/gov/hmcts/reform/roleassignment/domain/model/Assignment.java

```diff
@@ -48,4 +49,9 @@
     public void log(String message) {
         log = (log == null ? "" : log + "\n") + message;
     }
+
+    public void setAttribute(String key, String value) {
+        JsonNode valueNode = JacksonUtils.convertValueJsonNode(value);
```

Comment:
No need to create this new method.

Reply:
This new method is required and called in the iac-case drool rules.

Reply:
As suggested we can create the mutable map by new HashMap<>(Map.of())  then no need to define here.

Reply:
I think, it is fine. unless there is no issue with this approach.